### PR TITLE
feat: added authorization header to all requests

### PR DESCRIPTION
**Pull request by oskarikerppo**

### CODE REVIEW

File: lib/utils/fetch.ts [hunk 1]
```diff
@@ -1,6 +1,9 @@
 // https://eckertalex.dev/blog/typescript-fetch-wrapper
-async function http<T>(path: string, config?: RequestInit): Promise<T> {
-    const request = new Request(path, { headers: { 'Content-Type': 'application/json' }, ...config });
+async function http<T>(path: string, config?: RequestInit, auth?: string): Promise<T> {
+    const request = new Request(path, {
+        headers: { 'Content-Type': 'application/json', Authorization: `Bearer ${auth}` },
+        ...config,
+    });
```
Comment (LeadingMoominExpert) [inline lib/utils/fetch.ts:6]: Can't you just init the `const idJwt = useUser()?.getSignInUserSession()?.getIdToken().getJwtToken();` variable here and set the Authorization header if it's not null? Then you wouldn't need to get the token on every form and pass it along to this function

**Review by LeadingMoominExpert — CHANGES_REQUESTED** [review]

File: lib/utils/fetch.ts (hunk 1 above)
Comment (oskarikerppo) [reply lib/utils/fetch.ts:6]: `import { useUser } from '@/lib/hooks/useAuth';

// https://eckertalex.dev/blog/typescript-fetch-wrapper
async function http<T>(path: string, config?: RequestInit): Promise<T> {
    const idJwt = useUser()?.getSignInUserSession()?.getIdToken().getJwtToken();
    const request = new Request(path, {
        headers: { 'Content-Type': 'application/json', Authorization: `Bearer ${idJwt}` },
        ...config,
    });
    const response = await fetch(request);`

Doesn't seem to work

Comment (LeadingMoominExpert) [reply lib/utils/fetch.ts:6]: How about 
```typescript
import { Auth } from 'aws-amplify';
...
const authSession = await Auth.currentSession();
const jwt = authSession.getIdToken()?.getJwtToken();
```

Comment (oskarikerppo) [reply lib/utils/fetch.ts]: That works

File: components/projects/NewProject/NewCustomer.tsx [hunk 2]
```diff
@@ -37,6 +38,7 @@ function NewCustomer(): JSX.Element {
     });
     const url = `${process.env.NEXT_PUBLIC_API_URL}/customer`;
     const { mutate } = useSWRConfig();
+    const idJwt = useUser()?.getSignInUserSession()?.getIdToken().getJwtToken();
```
Comment (LeadingMoominExpert) [inline components/projects/NewProject/NewCustomer.tsx:41]: Now you can remove the token definitions and useUser imports from the components

**Review by LeadingMoominExpert — CHANGES_REQUESTED** [review]

**Review by LeadingMoominExpert — APPROVED** [review]

**Pull request merged**